### DISCUSSION TITLE
Playback and other events per note/beat

### DISCUSSION THREAD

**Discussion by stereokai** [CoderLine/alphaTab#755] [Q&A] [answered]
Are any of the following APIs available/planned?

1. User clicked on a note
2. User clicked on a beat
3. Highlight note
4. Highlight beat
5. Play a note by clicking
6. Play a beat by clicking

Thanks

**Comment by Danielku15** [comment, marked as answer]
As an overall summary answer: No, I don't directly plan to provide full UI level editor capabilities out of the box within alphaTab. This is something that needs to be build using the other provided APIs. 

1. No, not as built in feature currently. There is https://alphatab.net/docs/reference/settings/core/includenotebounds/ which allows you to obtain the individual note head position. Should be fairly easy to build something for your needs. 
2. Is already available. https://alphatab.net/docs/reference/api/beatmousedown/ https://alphatab.net/docs/reference/api/beatmousemove/ and https://alphatab.net/docs/reference/api/beatmouseup/ are mouse related events on beat interaction (fired during selection). 
3. No. 
4. No. 
5. https://alphatab.net/docs/reference/api/playnote/
6. https://alphatab.net/docs/reference/api/playbeat/

Generally I recommend that you look on the website for the documented APIs.

**Comment by stereokai** [comment]
Thanks for replying! I was familiar with `playnote` and `playbeat`, but I specifically meant if anything exists for playing a note or beat by clicking
